.github/workflows/pre-commit.yml

```diff
@@ -2,8 +2,9 @@ name: Run pre-commit
 
 on:
   push:
+    branches: [main]
   pull_request:
-    types: opened
+    branches: [main]
```

Comment:
1. for any push to main
2. for any PR event that targets main branch. Note, default event types for a PR is `[opened, sync, re-opened]`